src/background/stores/card-selection-store.ts

```diff
@@ -24,17 +24,72 @@ export class CardSelectionStore extends BaseStoreImpl<CardSelectionStoreData> {
 
     public getDefaultState(): CardSelectionStoreData {
         const defaultValue: CardSelectionStoreData = {
-            rules: null,
+            rules: {},
         };
 
         return defaultValue;
     }
 
-    public toggleRuleExpandCollapse(payload: RuleExpandCollapsePayload): void {}
+    private deselectAllCardsInRule = (rule: RuleExpandCollapseData): void => {
+        if (!rule) {
+            return;
+        }
 
-    public toggleCardSelection(payload: CardSelectionPayload): void {}
+        for (const resultInstanceUid of Object.keys(rule.cards)) {
+            rule.cards[resultInstanceUid] = false;
+        }
+    };
 
-    public CollapseAllRules(): void {}
+    public toggleRuleExpandCollapse = (payload: RuleExpandCollapsePayload): void => {
+        if (!payload || !payload.ruleId || !this.state.rules || !this.state.rules[payload.ruleId]) {
+            return;
+        }
 
-    private onScanCompleted = (payload: UnifiedScanCompletedPayload): void => {};
+        const rule = this.state.rules[payload.ruleId];
+
+        rule.isExpanded = !rule.isExpanded;
+
+        if (!rule.isExpanded) {
+            this.deselectAllCardsInRule(rule);
+        }
+
+        super.emitChanged();
+    };
+
+    public toggleCardSelection = (): void => {};
+
+    public CollapseAllRules = (): void => {
+        for (const ruleId of Object.keys(this.state.rules)) {
+            const rule = this.state.rules[ruleId];
+            rule.isExpanded = false;
+            this.deselectAllCardsInRule(rule);
+        }
+
+        super.emitChanged();
+    };
+
+    private onScanCompleted = (payload: UnifiedScanCompletedPayload): void => {
```

Comment:
In previous stores, we don't usually do a whole lot in the stores. If we are changing this pattern up here, which is fine, I'd like to come up with better descriptive names for the functions to explain what exactly they are doing. Since these private anyways, it's not going to matter to any consumers anyways, but would help in readability and understanding.

Reply:
I hesitate to make that change because the name is identical to the function in the unified result store. Since they both listen for the same action, I'd like to keep them aligned.

Reply:
That's fine; I do think that is not fully necessary since we understand due to line 22, what action this function is linked to (scanComplete). 

The only issue is, that as is, this is not easily parse-able as to what this code is doing. Another approach would be to move some of the code to a more appropriately named private function (which can identify what this code is actually doing).